pandas/core/groupby.py

```diff
@@ -1487,27 +1489,39 @@ def aggregate(self, values, how, axis=0):
                 raise NotImplementedError
             out_shape = (self.ngroups,) + values.shape[1:]
 
-        if is_numeric_dtype(values.dtype):
-            values = com.ensure_float(values)
-            is_numeric = True
-            out_dtype = 'f%d' % values.dtype.itemsize
-        else:
-            is_numeric = issubclass(values.dtype.type, (np.datetime64,
-                                                        np.timedelta64))
+        if (values.dtype.kind == 'i' or
+                _is_datetime_or_timedelta_dtype(values.dtype)):
+            values = values.astype('int64')
+        elif is_bool_dtype(values.dtype):
```

Comment:
Doing the `view('int64')` gives an error:

```
ValueError: new type not compatible with array.
```

...because we're going from an int8 to and int64 for the purposes of being able to use iNaT in the Cython functions.

I found this on the numpy documentation:
> For a.view(some_dtype), if some_dtype has a different number of bytes per entry than the previous dtype (for example, converting a regular array to a structured array), then the behavior of the view cannot be predicted just from the superficial appearance of a (shown by print(a)). It also depends on exactly how a is stored in memory. Therefore if a is C-ordered versus fortran-ordered, versus defined as a slice or transpose, etc., the view may give different results.

Is the `astype` ok?